code/modules/games/cards.dm

```diff
@@ -214,7 +224,7 @@
 	if(!discarding || !to_discard[discarding] || !usr || !src) return
 
 	var/datum/playingcard/card = to_discard[discarding]
-	qdel(to_discard)
+	to_discard.Cut()
```

Comment:
Yeah; I just intentionally passed our `qdel` a list and it hard deleted it---noticed, then made the edit.